src/StaticWebAssetsSdk/Targets/Microsoft.NET.Sdk.StaticWebAssets.Compression.targets

```diff
@@ -0,0 +1,184 @@
+  <PropertyGroup>
+
+    <GenerateStaticWebAssetsPublishManifestDependsOn Condition="'$(WasmBuildingForNestedPublish)' != 'true'">
+      $(GenerateStaticWebAssetsPublishManifestDependsOn);
+      _CompressFilesForPublish;
+    </GenerateStaticWebAssetsPublishManifestDependsOn>
+
+    <StaticWebAssetsPrepareForRunDependsOn>
+      $(StaticWebAssetsPrepareForRunDependsOn);
+      _CompressFiles;
+    </StaticWebAssetsPrepareForRunDependsOn>
+
+    <StaticWebAssetsPrepareForPublishDependsOn>
+      $(StaticWebAssetsPrepareForPublishDependsOn);
+      _CompressFilesForPublish;
+    </StaticWebAssetsPrepareForPublishDependsOn>
+
+    <ResolveStaticWebAssetsInputsDependsOn>
+      $(ResolveStaticWebAssetsInputsDependsOn);
+      _ResolveCompressedFiles;
+    </ResolveStaticWebAssetsInputsDependsOn>
+
+    <ResolvePublishStaticWebAssetsDependsOn>
+      $(ResolvePublishStaticWebAssetsDependsOn);
+      _ResolveCompressedFilesForPublish;
+    </ResolvePublishStaticWebAssetsDependsOn>
+
+  </PropertyGroup>
+
+  <Target
+    Name="_ResolveCompressedFiles"
+    DependsOnTargets="$(ResolveCompressedFilesDependsOn)">
+
+    <ItemGroup>
+      <_CompressedFileIncludePatternForBuild
+        Include="@(CompressionConfiguration)"
+        Condition="'%(Stage)' == 'Build'" />
+    </ItemGroup>
+
+    <ResolveCompressedAssets
+      CandidateAssets="@(StaticWebAsset)"
+      CompressionConfigurations="@(_CompressedFileIncludePatternForBuild)"
+      ExplicitAssets="@(AssetToCompress)"
+      OutputBasePath="$(IntermediateOutputPath)">
+
+      <Output
+        TaskParameter="AssetsToCompress"
+        ItemName="_CompressedStaticWebAssets" />
+
+    </ResolveCompressedAssets>
+
+    <ItemGroup>
+      <StaticWebAsset Include="@(_CompressedStaticWebAssets)" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target
+    Name="_ResolveCompressedFilesForPublish"
+    DependsOnTargets="$(ResolveCompressedFilesForPublishDependsOn)">
+
+    <ItemGroup>
+
+      <_CompressedFileIncludePatternForPublish
+        Include="@(CompressionConfiguration)"
+        Condition="'%(Stage)' == 'Publish'" />
+
+      <_CandidateAssetsForPublish
+        Include="@(StaticWebAsset)"
+        Condition="'%(AssetKind)' != 'Build'" />
+
+    </ItemGroup>
+
+    <ResolveCompressedAssets
+      CandidateAssets="@(_CandidateAssetsForPublish)"
+      CompressionConfigurations="@(_CompressedFileIncludePatternForPublish)"
+      ExplicitAssets="@(AssetToCompress)"
+      OutputBasePath="$(IntermediateOutputPath)">
+
+      <Output
+        TaskParameter="AssetsToCompress"
+        ItemName="_CompressedStaticWebAssetsForPublish" />
+
+    </ResolveCompressedAssets>
+
+    <ItemGroup>
+      <StaticWebAsset Include="@(_CompressedStaticWebAssetsForPublish->'%(FullPath)')" />
+      <FilesToCompress Remove="@(FilesToCompress)" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target
+    Name="_CompressFiles"
+    DependsOnTargets="$(CompressFilesDependsOn)"
+    BeforeTargets="_SplitStaticWebAssetsByCopyOptions">
```

Comment:
You should not need this if you use `GenerateComputedBuildStaticWebAssetsDependsOn` I believe